core/trino-main/src/main/java/io/trino/operator/scalar/ArrayVectorFunctions.java

```diff
@@ -73,11 +73,9 @@ public static double cosineDistance(@SqlType("array(double)") Block first, @SqlT
             secondMagnitude += secondValue * secondValue;
```

Comment:
Change the commit message to say “improve accuracy of cosine distance calculation” and add a short explanation in the body describing why it fixes it. Feel free to copy from my earlier comments.